utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java

```diff
@@ -265,32 +256,35 @@ private static DescriptorProto messageElementToDescriptorProto(
             //Fields are optional by default in Proto3.
             String label = fieldLabel != null ? fieldLabel.toString().toLowerCase() : OPTIONAL;
 
+            String fieldType = null;
             ProtoType protoType = field.getType();
             String fieldTypeName = String.valueOf(protoType);
             ProtoType keyType = protoType.getKeyType();
             ProtoType valueType = protoType.getValueType();
             // Map fields are only permitted in messages
             if (protoType.isMap() && keyType != null && valueType != null) {
                 label = "repeated";
-                fieldTypeName = toMapEntry(field.getName());
+                fieldType = "message";
+                String fieldMapEntryName = toMapEntry(field.getName());
+                fieldMapEntryName = fieldMapEntryName.substring(0, 1).toUpperCase() + fieldMapEntryName.substring(1);
```

Comment:
Can this throw OutOfBoundsOfException if `fieldMapEntryName` is one character?

Reply:
Also good to add a comment why we have to do this.

Reply:
OutOfBoundsException cannot be thrown because the maps entry name will always have "Entry" as the suffix.